### DISCUSSION TITLE
外部电话呼入如何取消验证

### DISCUSSION THREAD

**Discussion by jibinghu** [rts-cn/rts#51] [XSwitch]
请问外部电话呼入时候返回 407 Proxy Authentication Required的验证如何取消，希望把对方ip设置为白名单，对应的网关都已经配置了。

---

# 1. 呼叫初始化：来自 182.92.***.*** 的 INVITE 请求
recv 1122 bytes from udp/[182.92.***.***]:5089 to udp/[125.71.***.***]:7060 at 2024-11-26 13:28:08.425790:
INVITE sip:*******18458@125.71.***.***:7060 SIP/2.0
From: "*******9240" <sip:*******9240@182.92.***.***>;tag=38y4XSc1ZNU3j
Call-ID: 38549610-abb7-11ef-8b72-fd440bad93a2
User-Agent: uincall_sbc

# 2. FreeSWITCH 创建新会话
019366f0-f7ea-7473-975c-6625e296700e 2024-11-26 13:28:08.417813 [NOTICE] switch_channel.c:1244 
New Channel sofia/default/*******9240@182.92.***.*** [019366f0-f7ea-7473-975c-6625e296700e]

# 3. ACL 验证通过
2024-11-26 13:28:08.417813 [DEBUG] sofia.c:11013 
verifying acl "domains" for ip/port 182.92.***.***:0.

# 4. FreeSWITCH 要求设备进行认证，返回 407 Proxy Authentication Required
send 828 bytes from udp/[125.71.***.***]:7060 to udp/[182.92.***.***]:5089 at 2024-11-26 13:28:08.428309:
SIP/2.0 407 Proxy Authentication Required
Proxy-Authenticate: Digest realm="182.92.***.***", nonce="019366f0-f7eb-7520-9020-dca32639808e", algorithm=MD5, qop="auth"

# 5. 设备重新发起 INVITE 请求，包含 Proxy-Authorization 信息
recv 1399 bytes from udp/[182.92.***.***]:5089 to udp/[125.71.***.***]:7060 at 2024-11-26 13:28:08.459730:
INVITE sip:*******18458@125.71.***.***:7060 SIP/2.0
Proxy-Authorization: Digest username="not-used", realm="182.92.***.***", nonce="019366f0-f7eb-7520-9020-dca32639808e", response="be7641563f0a5d1dee50b63877273a23"

# 6. 用户或域名未定义，FreeSWITCH 返回警告
2024-11-26 13:28:08.457811 [WARNING] sofia_reg.c:3264 
Can't find user [not-used@xswitch.cn] from 182.92.***.***
You must define a domain called 'xswitch.cn' in your directory and add a user with the id="not-used" attribute.

# 7. FreeSWITCH 拒绝呼叫，返回 403 Forbidden
send 684 bytes from udp/[125.71.***.***]:7060 to udp/[182.92.***.***]:5089 at 2024-11-26 13:28:08.472716:
SIP/2.0 403 Forbidden
Call-ID: 38549610-abb7-11ef-8b72-fd440bad93a2
CSeq: 91745653 INVITE

# 8. 呼叫被拒绝，挂断原因：CALL_REJECTED
2024-11-26 13:28:08.477821 [NOTICE] sofia.c:2513 Hangup sofia/default/*******9240@182.92.***.*** [CS_NEW] [CALL_REJECTED]

# 9. 记录事件：通道被销毁，原因是认证失败（CALL_REJECTED）
request_body: {"Event-Name":"CHANNEL_DESTROY", "Hangup-Cause":"CALL_REJECTED", "Channel-Call-UUID":"019366f0-f7ea-7473-975c-6625e296700e"}

**Comment by seven1240** [comment]
如果是默认安装，让对方把呼叫送到 5080 端口上就行，external profile 在5080 上是不验证的。当然，你也可以把 internal 中 auth-calls 设为 false ，但是自己要注意安全。因为这样所有人都能呼入了。

**Comment by jibinghu** [comment]
已收到，谢谢。
